src/main/java/edu/ucsb/cs156/gauchoride/controllers/RiderApplicationController.java

```diff
@@ -0,0 +1,214 @@
+        )
+        {
+        
+        long milli = 60 * 60 * 24 * 1000;
+        long currentTime = new Date().getTime();
+        long dateOnly = (currentTime / milli) * milli;
+
+        Date date = new Date(dateOnly);
+
+
+        RiderApplication riderApplication = new RiderApplication();
+        riderApplication.setUserId(getCurrentUser().getUser().getId());
+      
+        riderApplication.setPerm_number(perm_number);
+        riderApplication.setCreated_date(date);
+        riderApplication.setDescription(description);
+        
+        RiderApplication savedRiderApplication = riderApplicationRepository.save(riderApplication);
+
+        return savedRiderApplication;
+
+    }
+
+    @Operation(summary = "Gets a list of all rider request owned by current user")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_USER') || hasRole('ROLE_DRIVER') || hasRole('ROLE_MEMBER')")
+    @GetMapping("/all")
+    public ResponseEntity<String> riderRequests()
+            throws JsonProcessingException {
+                Long userId = super.getCurrentUser().getUser().getId();
+                Iterable<RiderApplication> riderApplications = riderApplicationRepository.findByUserId(userId);
+                String body = mapper.writeValueAsString(riderApplications);
+                return ResponseEntity.ok().body(body);
+
+            }
+
+    @Operation(summary = "Gets a personal rider request by id")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_USER') || hasRole('ROLE_DRIVER') || hasRole('ROLE_MEMBER')")
+    @GetMapping("/get")
+    public RiderApplication riderApplicationID(
+            @Parameter(name = "id", description = "Long, id number ride to get ", example = "1", required = true) @RequestParam Long id)
+            throws JsonProcessingException {
+        RiderApplication riderApplication = riderApplicationRepository.findByIdAndUserId(id, getCurrentUser().getUser().getId())
+                .orElseThrow(() -> new EntityNotFoundException(RiderApplication.class, id));
+        return riderApplication;
+    }
+
+    @Operation(summary = "Update a personal application")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_USER') || hasRole('ROLE_DRIVER') || hasRole('ROLE_MEMBER')")
+    @PutMapping("put")
+    public RiderApplication updatePersonalApplication(@Parameter(name = "id", description = "Long, id number ride to get ", example = "1", required = true) @RequestParam long id, 
+    @RequestBody @Valid RiderApplication incoming
+    ){
+        RiderApplication riderApplication = riderApplicationRepository.findByIdAndUserId(id, getCurrentUser().getUser().getId())
+            .orElseThrow(() -> new EntityNotFoundException(RiderApplication.class, id));
+
+            long milli = 60 * 60 * 24 * 1000;
+            long currentTime = new Date().getTime();
+            long dateOnly = (currentTime / milli) * milli;
+
+            Date date = new Date(dateOnly);
```

Comment:
This probably should have been factored out into a utility method.  But since this is the last round, I'll let it slide.